iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -555,7 +555,9 @@ private async Task HandleIncomingMessagesAsync()
                 Logging.Enter(this, "Process C2D message via callback", nameof(HandleIncomingMessagesAsync));
 
             Message message = ProcessMessage(true);
-            await (_deviceMessageReceivedListener?.Invoke(message) ?? TaskHelpers.CompletedTask).ConfigureAwait(false);
```

Comment:
we don't need to await the completion of the user supplied callback, we can simply invoke it and move-on (we are not dependent on the completion of the callback to dispose any resources)

Reply:
Does it generate a warning that needs to be suppressed?

Reply:
a warning for...?

Reply:
unawaited async calls

Reply:
it does not, no

Reply:
we are invoking a function, which returns a Task. there are no warnings regarding us not awaiting the returned task